### ISSUE TITLE
Shortening the ViT model space

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/tfhub.dev#78]
@MorganR @kempy here is the search criteria I used to shorten the ViT model space - 

1. Top-1 accuracy on the ImageNet-1k `val` set should be at least 75%. 
2. Fileterd out the models having an input size of 224 x 224.
3. Then I reverse sorted all the models from 1) and took the top-10. 

The top-1 accuracies we got from these models are bounded within [77.002%, 79.836%]. The models (S/16) we get from the above have ~22 Million parameters. This is comparable to a standard Resnet50. 

Note that these models were first pre-trained on the ImageNet-21k dataset and then were fine-tuned on ImageNet-1k. When I applied the similar conditions on the models pre-trained on ImageNet-1k directly, the highest top-1 accuracy I got on the ImageNet-1k `val` set is **76.696%**. 

Let me know if these 10 models look good to you and I will then proceed with the rest of the steps. 

My model exploration notebook is available [here](https://colab.research.google.com/gist/sayakpaul/c1e1d4c5eb2960e642b47686676c59d1/scratchpad.ipynb).

**Comment by sayakpaul** [comment]
@MorganR @kempy bumping this.

**Comment by MorganR** [comment]
Are these models all the same shape (S/16)? It might be more interesting to publish one or two best performing models for each model size, so that users can choose the most appropriate one for their compute needs.

SGTM to focus on models with 224x224 input.

**Comment by sayakpaul** [comment]
I can do another 10 from B/16. 20 models are still good for me to manage alone but beyond that, it's a bit difficult. 

> It might be more interesting to publish one or two best performing models for each model size, so that users can choose the most appropriate one for their compute needs.

All of that said, I think we also need to keep in mind that we will be publishing (at least I will) a tutorial that shows the porting end-to-end. So users will always have that flexibility.

**Comment by sayakpaul** [comment]
@MorganR I was hoping to begin this work this weekend. So, how about I pick the best 5 S/16 and best 5 B/16 models keeping the resolution to 224x224? 

Or should I group the models by model family (B/16, S/16, L/16, etc.) and pick the top-5 or top-10 in total?

**Comment by MorganR** [comment]
I was thinking more along the lines of the second approach: group by family, and take the top 1 or two models per family, so that you publish about 10 models in total

**Comment by sayakpaul** [comment]
Models have been published: https://tfhub.dev/sayakpaul/collections/vision_transformer/1
